src/daemons/rhsmcertd.c

```diff
@@ -234,34 +236,97 @@ long long gen_random(long long max) {
     return random_num % true_max;
 }
 
-/* Handle program signals */
+/**
+ * Try to run Python script package-profile-uploader. This script tries to upload DNF profile
+ * to server, when server supports profile
+ * @return Return true, when uploading was successful. Otherwise, return false.
+ */
+static gboolean
+upload_package_profile ()
+{
+    int status = 0;
+
+    int pid = fork ();
+    if (pid < 0) {
+        error ("Fork failed");
+        exit (EXIT_FAILURE);
+    }
+    if (pid == 0) {
+        debug ("Uploading of package profile (executing: %s)", PACKAGE_PROFILE_UPLOADER);
+        execl (PACKAGE_PROFILE_UPLOADER, PACKAGE_PROFILE_UPLOADER_NAME, NULL);
+    }
+
+    waitpid (pid, &status, 0);
+    status = WEXITSTATUS (status);
+
+    if (status == 0) {
+        info ("Uploading of package profile performed successfully.");
+        // No need to repeat this action again
+        return true;
+    } else {
+        info ("Uploading of package profile terminated with status code: %d", status);
+        return false;
+    }
+}
+
+
+/**
+ * Handle program signals. This method can handle two signals SIGTERM and SIGUSR1
+ */
 void
 signal_handler(int signo) {
+    debug ("Handling signal %d", signo);
     if (signo == SIGTERM) {
+        info ("rhsmcertd is shutting down...");
         /* Close lock file and release lock on this file */
         if (fd_lock != -1) {
+            /* Truncate lock file to zero value (delete PID before shutting down) */
+            int ret;
+            ret = ftruncate(fd_lock, 0);
+            if (ret == -1) {
+                warn ("Unable to truncate file: %s, %s", LOCKFILE, strerror (errno));
+            }
```

Comment:
I don't want to remove the file. I want only to truncate the file.

Reply:
Hm can you please help me understand why truncating the file, rather than removing it?

Reply:
I don't want to do more changes than necessary. It would be good to improve locking in more ways. E.g. `/var/lock/subsys` is related to init system that we do not use anymore. I would deffer such changes to some other PR.

Reply:
I'm not sure I understand: the file truncation is introduced by this PR, it is not existing in rhsmcertd.c currently in `main`. Why was the truncation added _as part of this PR_? Which problem(s) does it solve/help about?

Reply:
Adding PID to this lock file was added as part of this PR. When `rhsmcertd` is not running, then lock file should not contain any PID. It should be empty. Truncating is easy way how to create empty file.

Reply:
When `rhsmcertd` is not running, there should be no lock file at all, not an empty file. Generally daemons/services that write pidfiles try to clean them as much as possible; usually "orphan" files are due to daemons/services suddenly crashing, or being killed with SIGKILL (which cannot be handled), or simply poor handling of the pidfile.

Reply:
I agree that there should not be no lock file, when `rhsmcertd` is not running, but it is current behavior. I'm sure that we should fix it in another PR.

Reply:
Considering you are changing this behaviour in this PR, then please make it right from the start, instead of adding a suboptimal way that needs to be changed later. All is needed here is to switch from `ftruncate()` to `unlink()`.

Reply:
I think we should address this now to not incur the extra time and process of another PR. @jirihnidek What's the reasoning as to why to wait?

Reply:
Done